### PR TITLE
GH-2027: Incorrect resolved import causes errors (only in open editor)

### LINKED ISSUE #2027: Incorrect resolved import causes errors (only in open editor)
When opening a file from an internal project (`OprAssert.n4js`) the import statement `import {inspect} from "util";` resolves to "lodash-es/util.js" instead of the project "util". When performing a build (no editor open) the import is resolved correctly.

### PR DESCRIPTION
**Pull request by mmews-n4**
closes #2027

### CODE REVIEW

File: tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/scoping/MultipleModulesScopingTest.xtend [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import org.junit.Test
+
+/**
+ * Tests for building yarn workspaces, in particular certain corner and special cases
+ * such as "yarn link".
+ */
+class MultipleModulesScopingTest extends AbstractIdeTest {
+
+	@Test
+	def void testTwoN4jsModulesWithSameName() throws Exception {
```
Comment (mor-n4) [inline tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/scoping/MultipleModulesScopingTest.xtend:23]: Nice tests and works everything as expected!

But you might wanna add these two tests for two special cases:
```xtend
	@Test
	def void testN4jsdAndJsModulesWithSameNameInSameSourceFolderOK() throws Exception {
		testWorkspaceManager.createTestProjectOnDisk(
			#[
				"Client" -> '''
					import {C} from "Module";
					C;
				''',
				"Module.n4jsd" -> '''
					export external const C;
				''',
				"Module.js" -> '''
					export const C = 2;
				'''
			]
		);
		startAndWaitForLspServer();
		assertIssues();
	}

	@Test
	def void testDynamicImportWarning() throws Exception {
		testWorkspaceManager.createTestProjectOnDisk(
			#[
				"Client" -> '''
					import * as M+ from "Module";
					M.C;
				''',
				"Module.n4jsd" -> '''
					export external const C;
				''',
				"Module.js" -> '''
					export const C = 2;
				'''
			]
		);
		startAndWaitForLspServer();
		assertIssues("Client" -> #["(Warning, [0:7 - 0:14], N4JSD module Module should not be imported dynamically.)"]);
	}
```

**Review by mor-n4 — APPROVED** [review]
Great to have this super important validation!

File: tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/scoping/MultipleModulesScopingTest.xtend (hunk 1 above)
Comment (mmews-n4) [reply tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/scoping/MultipleModulesScopingTest.xtend:23]: Thanks, I added the tests!

**Comment by mmews-n4** [comment]
Thanks!

**Pull request merged**